### ISSUE TITLE
6/22 UCF101 実行時のoutput.jsonの中身 

### ISSUE THREAD

**Issue by idekazuki** [idekazuki/diary#20]
git 上のコードを実行することはできた #19 が、何が出力されているかあまりまだわからない。
今回はoutput.json file の中身を解析する。

入力として使用したのはUCF-101のArcheryの動画。

v_Archery_g01_c01.avi
という動画で、約５ｓの長さがある。
実行時のコマンドライン出力画面は次のようになった。
```
python3 main.py --input ./input --video_root ../../UCF-101/Archery/ --output ./output.json --model ./pretrain_data/resnet-18-kinetics.pth --mode score --model_depth 18
loading model ./pretrain_data/resnet-18-kinetics.pth
>>

loading model ./pretrain_data/resnet-18-kinetics.pth
resnet-18
../../UCF-101/Archery/v_Archery_g01_c01.avi
ffmpeg version 2.6.3 Copyright (c) 2000-2015 the FFmpeg developers
  built with gcc 4.8 (Ubuntu 4.8.4-2ubuntu1~14.04)
  configuration: --prefix=/usr/local --enable-shared --disable-static
  libavutil      54. 20.100 / 54. 20.100
  libavcodec     56. 26.100 / 56. 26.100
  libavformat    56. 25.101 / 56. 25.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 11.102 /  5. 11.102
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  1.100 /  1.  1.100
Input #0, avi, from '../../UCF-101/Archery/v_Archery_g01_c01.avi':
  Metadata:
    encoder         : MEncoder r34118-4.2.5
  Duration: 00:00:05.24, start: 0.000000, bitrate: 781 kb/s
    Stream #0:0: Video: mpeg4 (Simple Profile) (XVID / 0x44495658), yuv420p, 320x240 [SAR 1:1 DAR 4:3], 669 kb/s, 25 fps, 25 tbr, 25 tbn, 25 tbc
    Stream #0:1: Audio: mp3 (U[0][0][0] / 0x0055), 44100 Hz, stereo, s16p, 99 kb/s
[swscaler @ 0xe3f200] deprecated pixel format used, make sure you did set range correctly
Output #0, image2, to 'tmp/image_%05d.jpg':
  Metadata:
    encoder         : Lavf56.25.101
    Stream #0:0: Video: mjpeg, yuvj420p(pc), 320x240 [SAR 1:1 DAR 4:3], q=2-31, 200 kb/s, 25 fps, 25 tbn, 25 tbc
    Metadata:
      encoder         : Lavc56.26.100 mjpeg
Stream mapping:
  Stream #0:0 -> #0:0 (mpeg4 (native) -> mjpeg (native))
Press [q] to stop, [?] for help
frame=  131 fps= 32 q=24.8 Lsize=N/A time=00:00:05.24 bitrate=N/A    
video:546kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
/host/space0/ide-k/out_git/video-classification-3d-cnn-pytorch/classify.py:25: UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  inputs = Variable(inputs, volatile=True)
```

output.json を以下のようなコードで表示する。

```
#coding:utf-8
import json


def main():
    with open('./output.json') as f:
        jsn = json.load(f)
    #読み込んだ形式を表示
    print(type(jsn))
    #整えた書式で全体を表示
    print(json.dumps(jsn, sort_keys=True, indent=4))
    #ラベルのみを抽出
    for jsn_dic in jsn:
        print(len(jsn_dic["clips"]))
        for i in range(len(jsn_dic["clips"])):
            print(jsn_dic["clips"][i]["label"])
if __name__ == "__main__":
    main()
```

以下のような構造になっていることがわかった。

```
<class 'list'>                                                                                             │  1 [{"video": "v_Archery_g01_c01.avi", "clips": [{"segment": [1, 16], "label": "archery", "scores": [1.7
[                                                                                                          │    758756875991821, 2.1774258613586426, -0.3326228857040405, -0.29544246196746826, -2.281000852584839, 7
    {                                                                                                      │    .768344402313232, 0.3198256194591522, -1.115415334701538, -2.7847046852111816, 0.28276023268699646, -
        "clips": [                                                                                         │    3.1307008266448975, -1.4592198133468628, -0.11600761115550995, 0.8592607975006104, -0.768841564655304
            {                                                                                              │    , -0.8403390049934387, -0.1674695611000061, -0.3636217713356018, 0.06561709940433502, -2.458683490753
                "label": "archery",                                                                        │    174, 1.1174076795578003, 1.9969531297683716, -3.195760488510132, -0.8408206105232239, 1.0247690677642
                "scores": [                                                                                │    822, -0.40334686636924744, -0.3428850769996643, -1.8218177556991577, -1.6184970140457153, 1.839611530
                    1.7758756875991821,
                               .
                               .
                               .
                    1.2808388471603394,
                    -1.083033561706543
                ],
                "segment": [
                    113,
                    128
                ]
            }
        ],
        "video": "v_Archery_g01_c01.avi"
    }
]

```
とりあえず"label"がアーチェリーになっていれば動作認識ができていることになる。
ざっと見"archery"と"playing tennis"の２つのクラスが出力されていることがわかった。

とりあえずpythonでlabelだけ抽出してみる。
archery
archery
archery
playing tennis
archery
playing tennis
archery
archery

このように８個のラベルが推測値として出力されていた。
正解していたのは８個中６個だったので、ここだけ見ると精度は０．７ちょい
frame=  131 fps= 32 q=24.8 Lsize=N/A time=00:00:05.24 bitrate=N/A    
main.py を見るとopt.sample_duration = 16
となっているので131//16=8 frameとなって結果と一致していることがわかる。
